server/controllers/documents.js

```diff
@@ -0,0 +1,204 @@
+import models from '../models';
+import jwt from 'jsonwebtoken';
+import config from '../config/config';
+
+const Documents = models.Documents;
+const Users = models.Users;
+
+const checkDocOwner = (docId, queryId) => {
+  if (parseInt(docId, 10) === parseInt(queryId, 10)) return true;
+};
+
+const checkUserRole = (roleId) => {
+  if (roleId === 1) return true;
+}; 
+
+const documentController = {
+  create(req, res) {
+    const userData = req.body;
+    const loggedInUser = req.user;
+    if( req.body.title === '' || req.body.content === '' 
+    || req.body.access === '') {
+      return res.status(400).send({ message: 'All fields must not be empty'});
+    }
+    if (!loggedInUser) {
+      return res.status(403).send({ message: 'Request denied'});
+    }
+
+    Users.findOne({
+      where: {
+        id: req.user.id
+      }
+    })
+      .then(user => {
+        Documents.create(Object.assign({}, userData, { ownerId: user.id }))
+          .then(() => res.status(200).send({ message: 'Document created successfully.' }))
+          .catch(error => res.status(500).send(error));
+      })
+      .catch(error => res.status(500).send(error));
+  },
+
+  getAll(req, res) {
+    const loggedInUserRoleId = req.user.roleId;
+    const restrictedUser = checkUserRole(loggedInUserRoleId);
```

Comment:
Can be better written as: `const isWriter = checkIfWriter(loggedInUserRoleId);` See related comment above.